HeterogeneousCore/AlpakaCore/interface/chooseDevice.h

```diff
@@ -1,15 +1,28 @@
 #ifndef HeterogeneousCore_AlpakaCore_interface_chooseDevice_h
 #define HeterogeneousCore_AlpakaCore_interface_chooseDevice_h
 
+#include "FWCore/ServiceRegistry/interface/Service.h"
 #include "FWCore/Utilities/interface/StreamID.h"
 #include "HeterogeneousCore/AlpakaInterface/interface/config.h"
 #include "HeterogeneousCore/AlpakaInterface/interface/devices.h"
 #include "HeterogeneousCore/AlpakaInterface/interface/traits.h"
+#include "HeterogeneousCore/AlpakaServices/interface/alpaka/AlpakaService.h"
 
 namespace cms::alpakatools {
 
   template <typename TPlatform, typename = std::enable_if_t<is_platform_v<TPlatform>>>
   alpaka::Dev<TPlatform> const& chooseDevice(edm::StreamID id) {
+    edm::Service<ALPAKA_TYPE_ALIAS(AlpakaService)> service;
```

Comment:
As a general pattern I think using any of the macros that effectively depend on `ALPAKA_ACCELERATOR_NAMESPACE` in altering behavior of code not enclosed in `ALPAKA_ACCELERATOR_NAMESPACE` is dangerous.

Written that, assuming this "CUDA-style framework" would have a relatively short lifetime (before being superseded by #39428), I could be ok with this in order to keep the migration work going. We just must not enable more than one `ALPAKA_ACCELERATOR_NAMESPACE` per `TPlatform` as long as this kind of code is around.

Reply:
@makortel sorry for the delay -- yes, I fully agree with your assessment.

I would suggest to merge this PR in order to clean up the example code, and them move to #39428 by the next couple of weeks.

Reply:
Ok